ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/headervalidationrules/BaseFeeMarketBlockHeaderGasPriceValidationRule.java

```diff
@@ -40,16 +40,22 @@ public boolean validate(final BlockHeader header, final BlockHeader parent) {
     try {
 
       // if this is the fork block, baseFee should be the initial baseFee
-      if (baseFeeMarket.isForkBlock(header.getNumber())) {
+      if (BaseFeeMarket.ValidationMode.INITIAL.equals(
+          baseFeeMarket.baseFeeValidationMode(header.getNumber()))) {
         return baseFeeMarket
             .getInitialBasefee()
             .equals(header.getBaseFee().orElseThrow(() -> MissingBaseFeeFromBlockHeader()));
+      } else if (BaseFeeMarket.ValidationMode.NONE.equals(
+          baseFeeMarket.baseFeeValidationMode(header.getNumber()))) {
+        // skip validation to allow sync when a zero baseFee chain has historical baseFees
+        return true;
```

Comment:
if ZeroBaseFeeMarket overrode getInitialBaseFee and always returned zero, we could leave this conditional as it was?

Reply:
The scenario solved by this conditional is when you have some baseFee present on your network's block (enabled by London) and then subsequently set `zeroBaseFee: true`. So your blocks' baseFee goes from empty -> non-zero -> 0

During syncing, this rule ends up trying to compare the expected base fee (zero, due to overridden computeBaseFee) with the actual one (non-zero)